Build fails on NixOS

With the following error:

```
error: builder for '/nix/store/rg0rzjxyfsif4ynrl5xn2x6z4iwar6jn-virtual-desktops-2.2.0.drv' failed with exit code 2;
       last 10 log lines:
       > g++ -O2 -shared -g -fPIC --no-gnu-unique -std=c++23 -Wall -Iinclude -DWLR_USE_UNSTABLE `pkg-config --cflags pixman-1 libdrm hyprland`   src/VirtualDesk.cpp src/VirtualDeskManager.cpp src/main.cpp src/sticky_apps.cpp src/utils.cpp -o virtual-desktops.so
       > src/VirtualDesk.cpp: In member function 'Layout VirtualDesk::generateCurrentMonitorLayout()':
       > src/VirtualDesk.cpp:177:41: warning: comparison of integer expressions of different signedness: 'int' and 'long unsigned int' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wsign-compare-Wsign-compare8;;]
       >   177 |     for (int i = vdeskFirstWorkspace; i < vdeskFirstWorkspace + monitors.size(); i++) {
       >       |                                       ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > src/main.cpp:5:10: fatal error: hyprland/src/desktop/Workspace.hpp: No such file or directory
       >     5 | #include <hyprland/src/desktop/Workspace.hpp>
       >       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > compilation terminated.
       > make: *** [Makefile:17: virtual-desktops.so] Error 1
       For full logs, run 'nix log /nix/store/rg0rzjxyfsif4ynrl5xn2x6z4iwar6jn-virtual-desktops-2.2.0.drv'.
error: 1 dependencies of derivation '/nix/store/ya43mb1iq7mzcn54jscyw7j4d20ql1qh-hm_hyprhyprland.conf.drv' failed to build
error: 1 dependencies of derivation '/nix/store/a2rq2zqs5ln3yb0mvdqw5n5pqi9h9f9z-activation-script.drv' failed to build
error: 1 dependencies of derivation '/nix/store/6z6g0bpzf0qawqr26r19ny5gaz7a1y1r-home-manager-generation.drv' failed to build
error: 1 dependencies of derivation '/nix/store/kw3nfmdzwac9c9ghzyg0xijzlhq81j3k-unit-home-manager-david.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/psygazl4snc89sglxjcq13d7zdzl3xs2-system-units.drv' failed to build
error: 1 dependencies of derivation '/nix/store/13f9ilvy4nvk4x8m2kvv7j8djpzzd5fc-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/73m9j97v5ibb89svg98117rzzhs5xxxb-nixos-system-idea-24.05.20240316.c75037b.drv' failed to build
```

The issue seems to be a change in Hyprland which was addressed in #27. On NixOS, the build still uses Hyprland v0.36.0 that has the  file `Workspace.cpp` at a different location. I suspect all is needed is a change in this line below to a more recent version of Hyprland.

https://github.com/levnikmyskin/hyprland-virtual-desktops/blob/801baf76fa6b0e0ffa89b4092e070963f249dc13/flake.nix#L5

Just to be sure, pinging @wiillou.

--- a/flake.nix
+++ b/flake.nix
@@ -2,7 +2,7 @@
   description = "A plugin for the Hyprland compositor, implementing virtual-desktop functionality.";
 
   inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
-  inputs.hyprland.url = "github:hyprwm/Hyprland/v0.36.0";
+  inputs.hyprland.url = "github:hyprwm/Hyprland/v0.37.0";
 
   outputs = { self, nixpkgs, hyprland }:
     let

--- a/flake.nix
+++ b/flake.nix
@@ -2,7 +2,7 @@
   description = "A plugin for the Hyprland compositor, implementing virtual-desktop functionality.";
 
   inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
-  inputs.hyprland.url = "github:hyprwm/Hyprland/v0.37.0";
+  inputs.hyprland.url = "github:hyprwm/Hyprland/v0.37.1";
 
   outputs = { self, nixpkgs, hyprland }:
     let

